File: 算法/代码随想录算法PDF/00调试源代码/406.根据身高重建队列.cpp
/*
 * @lc app=leetcode.cn id=406 lang=cpp
 *
 * [406] 根据身高重建队列
 */

// @lc code=start
class Solution {
public:
    static bool cmp(vector<int>& a, vector<int>& b) {
        if (a[0] == b[0]) return a[1] < b[1];
        return a[0] > b[0];
    }
    vector<vector<int>> reconstructQueue(vector<vector<int>>& people) {
        sort(people.begin(), people.end(), cmp);
        list<vector<int>> que;
        for (int i = 0; i < people.size(); i++) {
            int index = people[i][1];
            list<vector<int>>::iterator it = que.begin();
            while (index--) {
                it++;
            }
            que.insert(it, people[i]);
        }
        return vector<vector<int>>(que.begin(), que.end());
    }
};
// @lc code=end

File: 算法/代码随想录算法PDF/00调试源代码/406.根据身高重建队列.py
#
# @lc app=leetcode.cn id=406 lang=python3
#
# [406] 根据身高重建队列
#

# @lc code=start
class Solution:
    def reconstructQueue(self, people: List[List[int]]) -> List[List[int]]:
        # lambda返回的是一个元组：先按x[0]从高到低排序
        # 当-x[0](维度h）相同时，再根据x[1]（维度k）从小到大排序
        people.sort(key = lambda x : [-x[0], x[1]])
        que = []
        for p in people:
            que.insert(p[1], p)
        return que
# @lc code=end

